ledger/error/src/main/scala/com/daml/error/definitions/LedgerApiErrors.scala

```diff
@@ -224,363 +301,231 @@ object LedgerApiErrors extends LedgerApiErrorGroup {
-    )
-    @Resolution("Inspect the reason given and correct your application.")
-    object InvalidField
-        extends ErrorCode(id = "INVALID_FIELD", ErrorCategory.InvalidIndependentOfSystemState) {
-      case class Reject(_reason: String)(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = s"The submitted command has a field with invalid value: ${_reason}"
+      @Explanation(
+        "The transaction does not exist or the requesting set of parties are not authorized to fetch it."
+      )
+      @Resolution(
+        "Check the transaction id and verify that the requested transaction is visible to the requesting parties."
+      )
+      object Transaction
+          extends ErrorCode(
+            id = "TRANSACTION_NOT_FOUND",
+            ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
+          ) {
+
+        case class Reject(transactionId: String)(implicit loggingContext: ContextualizedErrorLogger)
+            extends LoggingTransactionErrorImpl(cause = "Transaction not found, or not visible.") {
+          override def resources: Seq[(ErrorResource, String)] = Seq(
+            (ErrorResource.TransactionId, transactionId)
           )
-    }
+        }
+      }
 
-    @Explanation(
-      "This error is emitted when a submitted ledger API command specifies an invalid deduplication period."
-    )
-    @Resolution(
-      "Inspect the error message, adjust the value of the deduplication period or ask the participant operator to increase the maximum."
-    )
-    object InvalidDeduplicationPeriodField
-        extends ErrorCode(
-          id = "INVALID_DEDUPLICATION_PERIOD",
-          ErrorCategory.InvalidGivenCurrentSystemStateOther,
-        ) {
-      case class Reject(_reason: String, _maxDeduplicationDuration: Duration)(implicit
-          loggingContext: ContextualizedErrorLogger
-      ) extends LoggingTransactionErrorImpl(
-            cause = s"The submitted command had an invalid deduplication period: ${_reason}"
+      @Explanation(
+        "The ledger configuration could not be retrieved. This could happen due to incomplete initialization of the participant or due to an internal system error."
+      )
+      @Resolution("Contact the participant operator.")
+      object LedgerConfiguration
+          extends ErrorCode(
+            id = "LEDGER_CONFIGURATION_NOT_FOUND",
+            ErrorCategory.InvalidGivenCurrentSystemStateResourceMissing,
           ) {
-        override def context: Map[String, String] =
-          super.context + ("max_deduplication_duration" -> _maxDeduplicationDuration.toString)
+
+        case class Reject()(implicit
+            loggingContext: ContextualizedErrorLogger
+        ) extends LoggingTransactionErrorImpl(
+              cause = "The ledger configuration is not available."
```

Comment:
```suggestion
              cause = "The ledger configuration could not be retrieved"
```

To make it more distinct from `ErrorCategory.TransientServerFailure/Code.UNAVAILABLE`